src/App.tsx

```diff
@@ -1,24 +1,217 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, {
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { Todo } from './types/Todo';
+import * as todoService from './api/todos';
+import { TodoRow } from './components/TodoRow';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoFooter } from './components/TodoFooter';
+import { getFilteredTodo } from './utils/getFilteredTodo';
+import { TodoStatus } from './types/TodoStatus';
+import { ErrorMessage } from './components/ErrorMessage';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [processingTodoIds, setProcessingTodoIds] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [
+    selectedStatus,
+    setSelectedStatus,
+  ] = useState<TodoStatus>(TodoStatus.All);
+  const [inputFocus, setInputFocus] = useState(false);
+
+  useEffect(() => {
+    todoService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+      });
```

Comment:
it would be great to have some kind of loading state for fetching all of our todos